### PR TITLE
Added SQL notebook IntelliSense

### PR DESCRIPTION
**Pull request by YurongHe**
Implemented the simple version intelliSense support as we discussed -- make the connection when the cell is active. If this doesn't look good, let's try the other complicate solutions.

Some notes:
- Didn't put connect into Kernel. Right now Kernel doesn't know active cell. also requestExecute is only string. But if we still need to put it in Kernel, we can definitely plumbing through those. Just SQL Kernel has lots of extra stuff comparing other Kernels.

- CellUri is not set until creating editor. so when addCell, uri is undefined, failed to connect.
  Move createUri to cell.ts and call it in cell constructor

- We have multiple places set activeCell. Should we consolidate them?

### CODE REVIEW

File: src/sql/parts/notebook/notebook.component.ts [hunk 1]
```diff
@@ -173,6 +175,14 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
+	private disconnect() {
+		if (this._model.defaultKernel.display_name === notebookConstants.SQL) {
+			if (this._model.activeCell && this._model.activeCell.cellType === CellTypes.Code && this._model.activeCell.cellUri) {
+				this.connectionManagementService.disconnect(this._model.activeCell.cellUri.toString());
```
Comment (kevcunnane) [inline src/sql/parts/notebook/notebook.component.ts]: Aren't these awaitable? You have to catch errors, even if swallow.

File: src/sql/parts/notebook/models/notebookModel.ts [hunk 2]
```diff
@@ -293,6 +293,12 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		// Set newly created cell as active cell
 		this.updateActiveCell(cell);
 
+		if (this._defaultKernel.display_name === notebookConstants.SQL
+			&& cell.cellType === CellTypes.Code
+			&& cell.cellUri) {
+			this.notebookOptions.connectionService.connect(this.activeConnection, cell.cellUri.toString());
```
Comment (kevcunnane) [inline src/sql/parts/notebook/models/notebookModel.ts]: Catch on the thenable?

**Comment by chlafreniere** [comment]
> to

#4065.

File: src/sql/parts/notebook/notebook.component.ts (hunk 1 above)
Comment (chlafreniere) [inline src/sql/parts/notebook/notebook.component.ts]: can we catch in a similar fashion to connect above?

**Pull request merged**